src/components/CountryCard/index.tsx

```diff
@@ -1,16 +1,55 @@
-import React from "react";
-
-interface CountryCardProps {}
+import React, { useContext } from "react";
+import Image from "next/image";
+import { ThemeContext } from "@/context/context";
+import { Theme } from "@mui/joy";
+interface CountryCardProps {
+  countryCode: string;
+  countryName: string;
+  countryPopulation: number;
+  countryRegion: string;
+  countryCapital: string[];
+  countryFlag: string;
+}
 
-function CountryCard() {
+function CountryCard({
+  countryCode,
+  countryName,
+  countryPopulation,
+  countryRegion,
+  countryCapital,
+  countryFlag,
+}: CountryCardProps) {
+  let currentTheme = useContext(ThemeContext);
   return (
-    <div>
-      <div></div>
-      <div>
-        <p></p>
-        <p></p>
-        <p></p>
-        <p></p>
+    <div className="app__country-card">
+      <div className="app__country-card_flag">
+        <img src={countryFlag} alt={countryName} />
+      </div>
+      <div
+        className={`app__country-card_info-section ${
+          currentTheme == "Dark Mode"
+            ? "container_dark"
+            : "container_light light"
+        }`}
+      >
+        <h2 className="app__country-card-title">{countryName}</h2>
+        <p>
+          <span>Population:</span> {countryPopulation}
+        </p>
+        <p>
+          <span>Region:</span> {countryRegion}
+        </p>
+        <p>
+          <span>Capital: </span>
+          {countryCapital
+            ? countryCapital.length > 1
+              ? `${countryCapital.forEach((capital, index) => {
+                  if (index != countryCapital.length - 1) return `${capital}, `;
```

Comment:
You can refactor this block of code by a small func
```
const formatCapitalsDisplay = (capitals: string[]): string => {
  if (!capitals || capitals.length === 0) return ""
  
  return capitals.join(', ')
}
```
```
<span>Capital: </span>
{formatCapitalsDisplay(countryCapital)}
```

Reply:
Okay thanks